packages/flutter/lib/src/widgets/binding.dart

```diff
@@ -399,30 +399,38 @@ mixin WidgetsBinding on BindingBase, SchedulerBinding, GestureBinding, RendererB
 
   @override
   void handleMetricsChanged() {
-    super.handleMetricsChanged();
-    for (WidgetsBindingObserver observer in _observers)
-      observer.didChangeMetrics();
+    afterWarmUpFrame(() {
```

Comment:
I think this won't be robust enough - we'll eventually add some new `handle` call, or another implementer of the binding would potentially dirty the tree in some unexpected way.  Can we somehow push this up to the RendererBinding?